pkg/cloudprovider/providers/vsphere/vclib/datastore.go

```diff
@@ -47,3 +47,22 @@ func (ds *Datastore) GetType(ctx context.Context) (string, error) {
 	}
 	return dsMo.Summary.Type, nil
 }
+
+// IsCompatibleWithStoragePolicy returns true if datastore is compatible with given storage policy else return false
+func (ds *Datastore) IsCompatibleWithStoragePolicy(ctx context.Context, storagePolicyId string) (bool, error) {
+	pbmClient, err := NewPbmClient(ctx, ds.Client())
+	if err != nil {
+		glog.Errorf("Failed to get new PbmClient Object. err: %v", err)
+		return false, err
+	}
+	datastores := []*Datastore{ds}
+	compatibleDatastore, faultMessages, err := pbmClient.GetCompatibleDatastores(ctx, storagePolicyId, datastores)
```

Comment:
Added new func 

// IsDataStoreCompatible check if the datastores is compatible for given storage policy id
// if datastore is not compatible with policy, fault message with the Datastore Name is returned
`func (pbmClient *PbmClient) IsDataStoreCompatible(ctx context.Context, storagePolicyID string, datastore *Datastore) (bool, string, error) {`